src/style-spec/expression/definitions/collator.js

```diff
@@ -0,0 +1,141 @@
+// @flow
+
+import { StringType, BooleanType, CollatorType } from '../types';
+
+import type { Expression } from '../expression';
+import type EvaluationContext from '../evaluation_context';
+import type ParsingContext from '../parsing_context';
+import type { Type } from '../types';
+
+declare var Intl: {
+  Collator: Class<Collator>
+}
+
+declare class Collator {
+  constructor (
+    locales?: string | string[],
+    options?: CollatorOptions
+  ): Collator;
+
+  static (
+    locales?: string | string[],
+    options?: CollatorOptions
+  ): Collator;
+
+  compare (a: string, b: string): number;
+
+  resolvedOptions(): any;
+}
+
+type CollatorOptions = {
+  localeMatcher?: 'lookup' | 'best fit',
+  usage?: 'sort' | 'search',
+  sensitivity?: 'base' | 'accent' | 'case' | 'variant',
+  ignorePunctuation?: boolean,
+  numeric?: boolean,
+  caseFirst?: 'upper' | 'lower' | 'false'
+}
+
+export class CollatorInstantiation {
```

Comment:
Suggested renames:
 - `declare class Collator` -> `declare class Intl$Collator` (this should be okay, since it's only being declared at top-level so that it can be referenced within the `Intl` class declaration, right?)
 - `CollatorInstantiation` -> `Collator`

Reply:
On a different note: `values.js` will need to become aware of this class, e.g., for `isValue`, `typeOf`

Reply:
I updated `values.js`, but I haven't actually figured out an expression test that would behave differently with the old `isValue => false` and `typeOf => ObjectType` behavior...